scripts/data_migrations/2018-10-31_migrate-data-sources.py

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env python
+
+import sys
+
+sys.path.insert(0, ".")
+
+from application import db
+from application.config import Config
+from application.factory import create_app
+from application.cms.models import Page, DataSource
+
+
+DATA_SOURCE_1_ATTRIBUTE_MAP = {
+    "source_text": "title",
+    "source_url": "source_url",
+    "type_of_data": "type_of_data",
+    "type_of_statistic_id": "type_of_statistic_id",
+    "department_source_id": "publisher_id",
+    "published_date": "publication_date",
+    "note_on_corrections_or_updates": "note_on_corrections_or_updates",
+    "frequency_id": "frequency_of_release_id",
+    "frequency_other": "frequency_of_release_other",
+    "data_source_purpose": "purpose",
+}
+DATA_SOURCE_2_ATTRIBUTE_MAP = {
+    "secondary_source_1_title": "title",
+    "secondary_source_1_url": "source_url",
+    "secondary_source_1_type_of_data": "type_of_data",
+    "secondary_source_1_type_of_statistic_id": "type_of_statistic_id",
+    "secondary_source_1_publisher_id": "publisher_id",
+    "secondary_source_1_date": "publication_date",
+    "secondary_source_1_note_on_corrections_or_updates": "note_on_corrections_or_updates",
+    "secondary_source_1_frequency_id": "frequency_of_release_id",
+    "secondary_source_1_frequency_other": "frequency_of_release_other",
+    "secondary_source_1_data_source_purpose": "purpose",
+}
+
+
+def get_page_attributes_collapsed_to_none(page, attrs_map):
+    data_source_attrs = {attr: getattr(page, attr, None) for attr in attrs_map.keys()}
+
+    for attr, value in data_source_attrs.items():
+        if value == "":
```

Comment:
Might it be worth checking for fields that are whitespace-only here too, as well as completely empty?

Reply:
Yep, done. 👍